[docs] Revise the dark mode article (#32179)
mui/material-ui: docs/data/material/customization/dark-mode/dark-mode.md

# Dark mode

<p class="description">Material UI comes with two palette modes: light (the default) and dark.</p>

## Dark mode by default

You can make your application use the dark theme by default setting the `mode: 'dark'`, regardless of the user's preference.

```js
import { ThemeProvider, createTheme } from '@mui/material/styles';

const darkTheme = createTheme({
  palette: {
    mode: 'dark',
  },
});

function App() {
  return (
    <ThemeProvider theme={darkTheme}>
      <main>This app is using the dark mode</main>
    </ThemeProvider>
  );
}

export default App;
```

While it seems like only a single value change, the `createTheme` helper actually modifies several palette values, adapting them accordingly to the dark mode.
The modified colors are the following:

{{"demo": "DarkTheme.js", "bg": "inline", "hideToolbar": true}}

> > Note: The colors are modified only if you use [the default palette](/customization/default-theme/). If you have a custom palette, you need to make sure that you have the correct values based on the `mode`. The following section explains how you can do it.

Reviewer: ```suggestion
> Note: The colors are modified only if you use [the default palette](/material-ui/customization/default-theme/). If you have a custom palette, you need to make sure that you have the correct values based on the `mode`. The following section explains how you can do it.
```

# Dark mode

<p class="description">Material UI comes with two palette modes: light (the default) and dark.</p>

## Dark mode by default

You can make your application use the dark theme by default setting the `mode: 'dark'`, regardless of the user's preference.

```js
import { ThemeProvider, createTheme } from '@mui/material/styles';

const darkTheme = createTheme({
  palette: {
    mode: 'dark',
  },

  function App() {
    return (
      <ThemeProvider theme={darkTheme}>
        <main>
          This app is using the dark mode
        </main>
      </ThemeProvider>
    );
}

export default App;

> ```

Reviewer: The code indent is wrong, it should be like this:

```js
import { ThemeProvider, createTheme } from '@mui/material/styles';

const darkTheme = createTheme({
  palette: {
    mode: 'dark',
  },
});

function App() {
  return (
    <ThemeProvider theme={darkTheme}>
      <main>
        This app is using the dark mode
      </main>
    </ThemeProvider>
  );
};

export default App;
```

## Toggling color mode

If you want to provide your users a way to toggle between modes, you can use React's context adding it to a button's `onClick` event:

{{"demo": "ToggleColorMode.js", "defaultCodeOpen": false}}

> ## System preference

Author: Uhm... "System preference" communicates better to me as in "this is how you get your app to follow/observe/match what either one of these 'systems' have as the defined mode". Does it make sense?

> ## System preference

Reviewer: I think `System preference` conveys better in this case. It guides how developers can achieve the `System preference`.

> ## System preference

Reviewer 2: Sounds good to me. 👍

> The method by which users express their mode preference can vary. It might be a system-wide setting exposed by the Operating System, or a setting controlled by the User Agent (user specifying a preference for a light or dark theme).

Reviewer 2: This is very wordy, and I think it can be pared down quite a bit. Is the end clause ("either systemwide, or for a single user agent") really a necessary detail? What I mean by that is: does the implementation change depending on how the preference is defined? Or is the same either way? If it's the same, I think we can omit that detail.

```suggestion
Users might have a preference for light or dark mode that they've set through their operating system—either systemwide, or for a single user agent.
```

> You can leverage this preference dynamically with the [useMediaQuery](/material-ui/react-use-media-query/) hook and the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query.

Reviewer 2: I really don't like the word "leverage" in this context but maybe that's just a pet peeve of mine. 😛

```suggestion
You can make use of this preference with the [useMediaQuery](/material-ui/react-use-media-query/) hook and the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query.
```

> For instance, you can enable dark mode automatically by checking if the user prefers it whether through the OS or browser settings:

Reviewer 2: ```suggestion
The following demo shows how to enable dark mode automatically by checking for the user's preference in their OS or browser settings:
```

## Toggling color mode

If you want to provide your users a way to toggle between modes, you can use React's context adding it to a button's `onClick` event:

{{"demo": "ToggleColorMode.js", "defaultCodeOpen": false}}

## System preference

The method by which users express their mode preference can vary. It might be a system-wide setting exposed by the Operating System, or a setting controlled by the User Agent (user specifying a preference for a light or dark theme).

You can leverage this preference dynamically with the [useMediaQuery](/components/use-media-query/) hook and the [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query.

> For instance, you can enable the dark mode automatically:

Author: This part here got me confused. What I understood from the first topic was that having the `createTheme` with `mode: 'dark'` would set the dark mode as the default one for the app (hence the title I added). But here it seems that we're talking about the same thing?! What is the actual difference?

Reviewer 2: If I'm understanding correctly:

- in the earlier section, it's talking about using dark mode as the default for the app, regardless of the user's preference
- in this part, it's showing how to check if the user prefers dark mode (whether through OS or browser settings), and then enabling/disabling based on that preference

I think you're right that we should do some more clarification.
